### DISCUSSION TITLE
Histogram

### DISCUSSION THREAD

**Discussion by nikolhaas65** [perspective-dev/perspective#2548] [Q&A] [answered]
Historgram is very common tool to visualize distributions. In provided examples with sales, it could be about comparing "sales" historgrams for  different "cities".  For realtime applications monitoring certain distributions is priceless.

PS. I just started using perspective, forgive me if histograms are implemented already. I just could not find them.
PSS. Quick self-reflection: perhaps, I can prepare histogram within DataFrame and then send it to Table/View

**Comment by texodus** [comment, marked as answer]
You can create a histogram by selecting an `"X Bar"` or `"Y Bar"` chart with your desired column in `group_by` field, and as (any) one column selected in the `columns` field with an aggregate of `"count"`. You can combine this with an expression column with the `bucket()` function as well, as in this [superstore](https://perspective.finos.org/block/?example=editable) `View` config:

```javascript
{
    "plugin": "X Bar",
    "group_by": [
        "Sales Bucket"
    ],
    "columns": [
        "Sales"
    ],
    "expressions": {
        "Sales Bucket": "bucket(\"Sales\",100)"
    },
    "aggregates": {"Sales": "count"}
}
```

https://github.com/finos/perspective/assets/60666/f12332a5-18c3-4fa5-8746-424279867b30

**Reply by nikolhaas65** [reply]
Thanks a lot.
